### PR TITLE
ConvNet class update

### PR DESCRIPTION
**Pull request by qbilius**
@yamins81 could you please review this (namely, model.py) and see if you like these changes?

Another purpose of this: practice PR for dldata :)

### CODE REVIEW

File: tfutils/model.py [hunk 1]
```diff
@@ -1,254 +1,216 @@
...
-    def conv(self, 
-             in_layer,
+        self.output = None
+        self._params = OrderedDict()
+
+    @property
+    def params(self):
+        return self._params
+
+    @params.setter
```
Comment (yamins81) [inline tfutils/model.py]: nice

File: tfutils/model.py [hunk 2]
```diff
@@ -1,254 +1,216 @@
...
              stride=1,
              padding='SAME',
+             init='xavier',
              stddev=.01,
              bias=1,
-             name=None):
+             activation='relu',
+             weight_decay=None,
+             in_layer=None):
+        if in_layer is None: in_layer = self.output
```
Comment (yamins81) [inline tfutils/model.py]: I prefer multi-line if's to these single-line statements, but am not wedded to it.

File: tfutils/model.py [hunk 3]
```diff
@@ -1,254 +1,216 @@
...
-                              padding=padding,
-                              name=name)
-
-    def print_activations(self, t):
-        print(t.op.name, ' ', t.get_shape().as_list())
+           activation='relu',
+           dropout=.5,
+           in_layer=None):
+        if in_layer is None: in_layer = self.output
+        resh = tf.reshape(in_layer,
```
Comment (yamins81) [inline tfutils/model.py]: right, this is a good place to put this

File: tfutils/model.py [hunk 4]
```diff
@@ -1,254 +1,216 @@
...
-                                   'alpha': alpha,
-                                   'beta': beta}
-        return tf.nn.lrn(in_layer, 
-                         depth_radius=depth_radius, 
-                         bias=bias, 
-                         alpha=alpha, 
-                         beta=beta,
-                         name=name)
+        kernel = tf.get_variable(initializer=self.initializer(init, stddev=stddev),
+                                 shape=[ksize, ksize, in_shape, out_shape],
```
Comment (yamins81) [inline tfutils/model.py]: do you think there's an argument for allowing non-square kernels?   For example:
    if ksize is an int:
        ksize1 = ksize2
    else
        ksize1, ksize2 = ksize

   and then call 
        kernel = tf.get_variable(...., shape=[ksize1, ksize2, ...])

And similarly for pooling?

File: tfutils/model.py (hunk 2 above)
Comment (qbilius) [inline tfutils/model.py]: I try to keep them single-line when they're really short and simple like here.

**Comment by qbilius** [comment]
I didn't realize you could comment so easily each line on PRs. Love this.

File: tfutils/model.py (hunk 2 above)
Comment (yamins81) [inline tfutils/model.py]: maybe we should make it consistent across the code base then.  (In my code in base.py I followed the other convention)

File: tfutils/model.py (hunk 4 above)
Comment (yamins81) [inline tfutils/model.py]: I may need this -- or rather, if it were done here I will probably at some point in the not too far future use this.

**Comment by qbilius** [comment]
See my latest changes 99d2c3d. If that's all, please merge the PR.
I'll change the if and other styling separately in the future.

**Pull request merged**